packages/serialize-request/lib/index.js

```diff
@@ -154,16 +154,33 @@ function serializeRequest(request, options = {}) {
  * Serialize request headers.
  *
  * @private
- * @param {Record<string, any>} headers
+ * @param {Headers | Record<string, any> | Iterable<[string, string]>} headers
  *     The headers object to serialize.
  * @param {Array<string>} includeHeaders
  *     An array of request headers to include.
  * @returns {SerializedRequestHeaders}
  *     Returns the serialized request headers.
  */
 function serializeHeaders(headers, includeHeaders) {
+	const headersObject = {};
+
+	if (Array.isArray(headers)) {
+		for (const [header, value] of headers) {
+			headersObject[header.toLowerCase()] = value;
+		}
+	} else if (typeof headers[Symbol.iterator] === 'function') {
+		// @ts-ignore
+		for (const [header, value] of headers) {
+			headersObject[header.toLowerCase()] = value;
+		}
```

Comment:
**suggestion:** it's a little more code but we can remove this `@ts-ignore` by defining a type checking function and using that in this condition. This will make sure we know we're being type-safe. If you change this line to something like:

```suggestion
	} else if (isIterableHeadersObject(headers)) {
		// @ts-ignore
		for (const [header, value] of headers) {
			headersObject[header.toLowerCase()] = value;
		}
```

Then we can define our type checking function like this and TypeScript will understand that, within that condition, the value has an iterator:

```js
/**
 * @param {any} value
 *     The value to test.
 * @returns {value is Iterable<string, string>}
 *     Returns whether a value is iterable.
 */
function isIterableHeadersObject(value) {
	return typeof value?.[Symbol.iterator] === 'function';
}
```